front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -58,6 +75,9 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [fetchedTrainSchedulesResults]
   );
 
+  const pacedTrainsResults = useMemo(() => fetchedPacedTrains || [], [fetchedPacedTrains]);
```

Comment:
Hm you can replace pacedTrainResults by these formatted trains, and this way reuse it below, for instance in useEffects (same for trainScheduleResults)
```suggestion
  const pacedTrains: PacedTrainResultWithPacedTrainId[] = useMemo(
    () =>
      showPacedTrains
        ? (fetchedPacedTrains || []).map((trainSchedule) => ({
            ...trainSchedule,
            id: formatEditoastTrainIdToPacedTrainId(trainSchedule.id),
          }))
        : [],
    [showPacedTrains, fetchedPacedTrains]
  );
```

Reply:
Done